### PR TITLE
Add new benchmarks and benchmarker functionality (try 2)

### PR DESCRIPTION
**Pull request by rctcwyvrn**
Fixes since the last PR

- ~~Use `ISO8601FormatStyle` to fix the linux build failure~~ It turns out that doesn't exist on corelibs-foundation either. It turns out the entire `format` function is missing so work around it with some `#if`
- Remove diff filtering for comparison results (just make sure it's more than a picosecond to remove any diffs from floating point rounding)
- Add a `createBenchmark.py` util script to create benchmarks + register them. After looking around at swift-collection-benchmark and the main swift repo's benchmarks, this seems to be the best way of doing things

### CODE REVIEW

**Comment by rctcwyvrn** [comment]
@swift-ci test

**Comment by rctcwyvrn** [comment]
@swift-ci test

File: Utils/createBenchmark.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# python3 createBenchmark.py MyRegexBenchmark
```
Comment (milseman) [inline Utils/createBenchmark.py:1]: If the benchmarks just added themselves to an array, would that simplify the registration process?

**Review by milseman — APPROVED** [review]
LGTM

**Comment by rctcwyvrn** [comment]
@swift-ci test

**Pull request merged**